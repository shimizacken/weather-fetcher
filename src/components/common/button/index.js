export { Button } from './Button';
export { ButtonTypes } from './buttonTypes';
