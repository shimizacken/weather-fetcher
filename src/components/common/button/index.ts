export { Button } from './Button';
export { ButtonType } from './buttonType';
export { BackButton } from './BackButton';
