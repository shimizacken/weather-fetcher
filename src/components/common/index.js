export { Button } from './button/Button';
export { SearchBox } from './searchBox/SearchBox';
export { RadioButton } from './radioButton/RadioButton';
export { Loader } from './loader/Loader';
export { Row } from './table';
export { Modal, ConfirmationModal } from './modal';
