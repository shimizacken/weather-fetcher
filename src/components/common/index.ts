export {Button} from './button/Button';
export {ButtonType} from './button/buttonType';
export {SearchBox} from './searchBox/SearchBox';
export {RadioButton} from './radioButton/RadioButton';
export {Loader} from './loader/Loader';
export {Table, Row, Column} from './table';
export {Modal, ConfirmationModal} from './modal';
export {BackButton} from './button';
export {TextInput} from './textInput/TextInput';
export {Logo} from './logo/Logo';
