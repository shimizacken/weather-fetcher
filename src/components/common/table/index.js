export { Row } from './row/Row';
