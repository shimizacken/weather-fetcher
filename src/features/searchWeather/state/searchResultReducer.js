import { SET_WEATHER } from './constants';

const initialState = {};

export const searchResult = (state = initialState, action) => {
  if (action.type === SET_WEATHER) {
    return {
      ...state,
      weather: action.weather
    };
  }

  return state;
};
