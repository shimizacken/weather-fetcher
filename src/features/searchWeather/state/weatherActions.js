import {
  SET_WEATHER,
  SET_WEATHER_END,
  FETCH_WEATHER_PENDING,
  FETCH_WEATHER_ERROR,
  FETCH_WEATHER_SUCCESS,
  SET_FETCH_WEATHER_FLAG
} from './constants';

export const setWeather = weather => ({
  type: SET_WEATHER,
  weather
});

export const setWeatherEnd = {
  type: SET_WEATHER_END
};

export const fetchWeatherPending = {
  type: FETCH_WEATHER_PENDING
};

export const fetchWeatherSuccess = weather => ({
  type: FETCH_WEATHER_SUCCESS,
  weather
});

export const fetchWeatherError = {
  type: FETCH_WEATHER_ERROR
};

export const setWeatherFetchFlag = isPending => ({
  type: SET_FETCH_WEATHER_FLAG,
  isPending
});

export const fetchWeather = url => dispatch => {
  dispatch(fetchWeatherPending);
  dispatch(setWeatherFetchFlag(true));

  return fetch(url)
    .then(response => response.json())
    .then(data => {
      if (data && data.cod !== 200) {
        throw data.message;
      }

      dispatch(setWeather(data));
      dispatch(setWeatherEnd);
    })
    .catch(error => {
      console.error(error);

      dispatch(fetchWeatherError);
    })
    .finally(() => dispatch(setWeatherFetchFlag(false)));
};
