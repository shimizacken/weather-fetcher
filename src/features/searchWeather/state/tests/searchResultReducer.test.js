/* eslint-disable no-undef */
import { randomAction } from 'app/state/common/randomAction';
import { searchResult } from '../searchResultReducer';
import { setWeather } from '../weatherActions';
import { searchResult as weatherMock } from './mock';

describe('searchResultReducer test suite', () => {
  it('should return initial state', () => {
    const action = randomAction();

    expect(searchResult(undefined, action)).toMatchSnapshot();
  });

  it('should return new state', () => {
    const action = setWeather(weatherMock);

    expect(searchResult(undefined, action)).toMatchSnapshot();
  });
});
