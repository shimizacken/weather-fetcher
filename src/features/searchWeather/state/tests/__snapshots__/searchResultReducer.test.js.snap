// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchResultReducer test suite should return initial state 1`] = `Object {}`;

exports[`searchResultReducer test suite should return new state 1`] = `
Object {
  "weather": Object {
    "payload": Object {
      "base": "stations",
      "clouds": Object {
        "all": 0,
      },
      "cod": 200,
      "coord": Object {
        "lat": 59.91,
        "lon": 10.75,
      },
      "dt": 1584810947,
      "id": 3143244,
      "main": Object {
        "feels_like": 0.25,
        "humidity": 30,
        "pressure": 1036,
        "temp": 4.14,
        "temp_max": 6,
        "temp_min": 1.67,
      },
      "name": "Oslo",
      "sys": Object {
        "country": "NO",
        "id": 1624,
        "sunrise": 1584767677,
        "sunset": 1584812009,
        "type": 1,
      },
      "timezone": 3600,
      "visibility": 10000,
      "weather": Array [
        Object {
          "description": "clear sky",
          "icon": "01d",
          "id": 800,
          "main": "Clear",
        },
      ],
      "wind": Object {
        "speed": 1,
      },
    },
    "type": "weather/fetch_success",
  },
}
`;
